### PR TITLE
add MongoDB 3.2 to the test matrix

### PR DESCRIPTION
**Pull request by mossprescott**
SD-1141 #done

Note: A couple of queries are not matching, so this isn't ready to go yet.

### CODE REVIEW

**Comment by mossprescott** [comment]
One test was failing just because the query was unsorted but the test expected a particular result, so that's fixed. Now there's one more failure that I haven't been able to figure out.

**Comment by sellout** [comment]
I see a few tests pending – are they related to field ordering? SD-1174 proposes being able to mark a backend as “unordered” (like “pending”) if it doesn’t respect field ordering.

**Comment by mossprescott** [comment]
I don't remember what's keeping each of those tests in pending (they've mostly been around for quite a while), but my sense is that it's mostly unresolved Mongo weirdness, like projecting `_id` and the behavior around missing values.

**Comment by sellout** [comment]
Ah, right – those are pending everywhere, not just 3.2

**Comment by mossprescott** [comment]
Why don't we just fix the field ordering, fer crissakes? We just need to insert a final (pair of) project(s), similar to what we do to eliminate _id. Er, unless we want to stream from a map-reduce, I guess, but in that case we should do it in memory.

**Comment by jdegoes** [comment]
> Why don't we just fix the field ordering, fer crissakes? 

Yeah, let's just bite the bullet. It would be ideal if we can detect some cases when this is not needed, just to avoid the performance penalty, but that's not strictly required.

**Comment by mossprescott** [comment]
@jedesah, you got this one just because I don't think I've given you anything recently.

File: scripts/build [hunk 1]
```diff
@@ -45,15 +46,18 @@ if [[ ${LOCAL_MONGODB:-} == "true" ]] ; then
   if [[ "$QUASAR_MONGODB_RELEASE" == "2_6" ]]; then
     export QUASAR_MONGODB_2_6="{\"mongodb\": {\"connectionUri\": \"mongodb://${QUASAR_MONGODB_HOST_2_6}\"}}"
   elif [[ "$QUASAR_MONGODB_RELEASE" == "3_0" ]]; then
-    export QUASAR_MONGODB_3_0="{\"mongodb\": {\"connectionUri\": \"mongodb://${QUASAR_MONGODB_HOST_3_0}\"}}"
+    export QUASAR_MONGODB_3_0="{\"mongodb\": {\"connectionUri\": \"mongodb://quasar-dbOwner:quasar@${QUASAR_MONGODB_HOST_3_0}/${QUASAR_MONGODB_TESTDB}\"}}"
```
Comment (jedesah) [inline scripts/build]: What's this `quasar-dbOwner`?

**Comment by jedesah** [comment]
I know this was not added by this PR, but what is Mongo 3.0 RO?

**Comment by jedesah** [comment]
I am under the impression this PR is "in progress" as opposed to "in review", but @mossprescott let me know if my assumption is wrong.

**Comment by wemrysi** [comment]
Mongo 3.0 RO is a read-only connection to a mongodb 3.0.x instance used to test that we can execute a subset of queries using the streaming functionality on a read-only connection.

**Comment by jedesah** [comment]
Ah, duh, should have figured that one out on my own.

**Comment by mossprescott** [comment]
@jedesah: this was meant to be ready, but now it's failing in Travis and it looks like I need to do a series of test builds, so I'll close this and open a new PR when I have it working.

**Pull request closed without merging**